test-clients/src/main/java/com/hedera/services/bdd/spec/queries/meta/HapiGetExecTime.java

```diff
@@ -0,0 +1,151 @@
+package com.hedera.services.bdd.spec.queries.meta;
+
+
+/*-
+ * ‌
+ * Hedera Services Test Clients
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.common.base.MoreObjects;
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.queries.HapiQueryOp;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.NetworkGetExecutionTimeQuery;
+import com.hederahashgraph.api.proto.java.NetworkGetExecutionTimeResponse;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionID;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+import static com.hedera.services.bdd.spec.queries.QueryUtils.answerCostHeader;
+import static com.hedera.services.bdd.spec.queries.QueryUtils.answerHeader;
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class HapiGetExecTime extends HapiQueryOp<HapiGetExecTime> {
+	private static final Logger log = LogManager.getLogger(HapiGetExecTime.class);
+
+	private List<TransactionID> txnIdsOfInterest;
+	private NetworkGetExecutionTimeResponse timesResponse;
+
+	private final List<String> txnsOfInterest;
+
+	public HapiGetExecTime(List<String> txnsOfInterest) {
+		this.txnsOfInterest = txnsOfInterest;
+	}
+
+	@Override
+	public HederaFunctionality type() {
+		return HederaFunctionality.NetworkGetExecutionTime;
+	}
+
+	@Override
+	protected HapiGetExecTime self() {
+		return this;
+	}
+
+	@Override
+	protected void assertExpectationsGiven(HapiApiSpec spec) throws Throwable {
+	}
+
+	@Override
+	protected void submitWith(HapiApiSpec spec, Transaction payment) {
```

Comment:
How am I supposed to access the times in the EET tests?  Rather than read the logs I was hoping to fail the test if the execution exceeded the safe threshold (1s? 300ms?), indicating we can be attacked.

Reply:
Have updated the `HapiGetExecTime` EET op with an `assertingNoneLongerTime()` method,
```
...
	getExecTime(cryptoTransfer, submitMessage, contractCall)
			.payingWith(GENESIS)
			.assertingNoneLongerThan(1, ChronoUnit.MILLIS)

```
that, when the max time is exceeded, in a failure such as,
```
2021-09-08 15:10:19.642 WARN   215  HapiSpecOperation - 'execTimesDemo' - HapiGetExecTime{} failed - {}
org.opentest4j.AssertionFailedError: Transaction 'contractCall' took PT0.043862991S, max allowed was PT0.001S! ==> expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55) ~[junit-jupiter-api-5.7.2.jar:5.7.2]
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40) ~[junit-jupiter-api-5.7.2.jar:5.7.2]
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:193) ~[junit-jupiter-api-5.7.2.jar:5.7.2]
	at com.hedera.services.bdd.spec.queries.meta.HapiGetExecTime.assertExpectationsGiven(HapiGetExecTime.java:87) ~[classes/:?]
```